ndonnx/_typed_array/masked_onnx.py

```diff
@@ -0,0 +1,881 @@
+        return DTypeInfoV1(author="ndonnx", type_name="NBoolean", meta=None)
+
+
+class NInt8(_NNumber):
+    _unmasked_dtype = onnx.int8
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayInt8:
+        return TyMaArrayInt8(data, mask)
+
+
+class NInt16(_NNumber):
+    _unmasked_dtype = onnx.int16
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayInt16:
+        return TyMaArrayInt16(data, mask)
+
+
+class NInt32(_NNumber):
+    _unmasked_dtype = onnx.int32
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayInt32:
+        return TyMaArrayInt32(data, mask)
+
+
+class NInt64(_NNumber):
+    _unmasked_dtype = onnx.int64
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayInt64:
+        return TyMaArrayInt64(data, mask)
+
+
+class NUInt8(_NNumber):
+    _unmasked_dtype = onnx.uint8
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayUInt8:
+        return TyMaArrayUInt8(data, mask)
+
+
+class NUInt16(_NNumber):
+    _unmasked_dtype = onnx.uint16
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayUInt16:
+        return TyMaArrayUInt16(data, mask)
+
+
+class NUInt32(_NNumber):
+    _unmasked_dtype = onnx.uint32
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayUInt32:
+        return TyMaArrayUInt32(data, mask)
+
+
+class NUInt64(_NNumber):
+    _unmasked_dtype = onnx.uint64
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayUInt64:
+        return TyMaArrayUInt64(data, mask)
+
+
+class NFloat16(_NNumber):
+    _unmasked_dtype = onnx.float16
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayFloat16:
+        return TyMaArrayFloat16(data, mask)
+
+
+class NFloat32(_NNumber):
+    _unmasked_dtype = onnx.float32
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayFloat32:
+        return TyMaArrayFloat32(data, mask)
+
+
+class NFloat64(_NNumber):
+    _unmasked_dtype = onnx.float64
+
+    def _build(
+        self, data: onnx.TyArray, mask: onnx.TyArrayBool | None
+    ) -> TyMaArrayFloat64:
+        return TyMaArrayFloat64(data, mask)
+
+
+# Nullable Singleton instances
+nbool = NBool()
+
+nfloat16 = NFloat16()
+nfloat32 = NFloat32()
+nfloat64 = NFloat64()
+
+nint8 = NInt8()
+nint16 = NInt16()
+nint32 = NInt32()
+nint64 = NInt64()
+
+nuint8 = NUInt8()
+nuint16 = NUInt16()
+nuint32 = NUInt32()
+nuint64 = NUInt64()
+
+nutf8 = NUtf8()
+
+# Union types
+#
+# Union types are exhaustive and don't create ambiguities with respect to user-defined subtypes.
+# TODO: Rename
+NCoreIntegerDTypes = (
```

Comment:
NPrimitiveIntegerDTypes?

Reply:
This now uses the same names as in `onnx.py` since the information is already in the module name.